sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -592,11 +572,18 @@ protected void initializeDeploymentPlanManager() {
      */
     protected void initializeRecoveryPlanManager() {
         LOGGER.info("Initializing recovery plan...");
-        LaunchConstrainer launchConstrainer = new TimedLaunchConstrainer(Duration.ofMillis(destructiveRecoveryDelayMs));
-        FailureMonitor failureMonitor =
-                permanentFailureTimeoutMs.isPresent()
-                        ? new TimedFailureMonitor(Duration.ofMillis(permanentFailureTimeoutMs.get()))
-                        : new NeverFailureMonitor();
+        LaunchConstrainer launchConstrainer;
+        FailureMonitor failureMonitor;
+        if (failurePolicyOptional.isPresent()) {
+            ReplacementFailurePolicy failurePolicy = failurePolicyOptional.get();
+            launchConstrainer = new TimedLaunchConstrainer(
+                    Duration.ofMillis(failurePolicy.getMinReplaceDelayMins()));
```

Comment:
`null` was being used as a signal to disable automated permanent failure recovery.  No longer necessary.